### DISCUSSION TITLE
%E4%B8%AD%E9%97%B4%E4%BB%B6/rocketmq/%E6%A0%B8%E5%BF%83%E7%9F%A5%E8%AF%86%E7%82%B9/RocketMQ%E4%B8%BA%E4%BB%80%E4%B9%88%E6%80%A7%E8%83%BD%E4%B8%8D%E5%A6%82Kafka

### DISCUSSION THREAD

**Discussion by giscus** [xiaobaiTech/golang-full-stack#22] [Announcements]
# %E4%B8%AD%E9%97%B4%E4%BB%B6/rocketmq/%E6%A0%B8%E5%BF%83%E7%9F%A5%E8%AF%86%E7%82%B9/RocketMQ%E4%B8%BA%E4%BB%80%E4%B9%88%E6%80%A7%E8%83%BD%E4%B8%8D%E5%A6%82Kafka

elasticSearch 是什么？工作原理是怎么样的？ 现在有三段文本，id 分别是 0、1、2，你需要快速找到哪段文本里含有关键词

https://golangguide.top/%E4%B8%AD%E9%97%B4%E4%BB%B6/rocketmq/%E6%A0%B8%E5%BF%83%E7%9F%A5%E8%AF%86%E7%82%B9/RocketMQ%E4%B8%BA%E4%BB%80%E4%B9%88%E6%80%A7%E8%83%BD%E4%B8%8D%E5%A6%82Kafka.html

**Comment by 0xiaobao0** [comment]
大师，我悟了

**Comment by xdydx** [comment]
你这里说的是 消费消息 的时候，Kafka 用的是 sendfile，RocketMQ 用的是 mmap，对于生产消息，二者用的都是 mmap
